crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -747,9 +747,8 @@ impl Stage for SystemStage {
             self.world_id = Some(world.id());
         }
 
-        if let ShouldRun::No | ShouldRun::NoAndCheckAgain =
-            self.stage_run_criteria.should_run(world)
-        {
+        let mut stage_should_run = self.stage_run_criteria.should_run(world);
+        if let ShouldRun::No | ShouldRun::NoAndCheckAgain = stage_should_run {
```

Comment:
Doesn't returning here on NoAndCheckAgain mean we don't check again?

Reply:
Yes: if we check again without running anything (which a `No*` implies) we're stuck in a loop.